packages/firestore/src/core/query.ts

```diff
@@ -27,158 +27,103 @@ import {
 import { FieldPath, ResourcePath } from '../model/path';
 import { assert, fail } from '../util/assert';
 import { Code, FirestoreError } from '../util/error';
-import { isNullOrUndefined } from '../util/types';
+import { Target } from './target';
 
+/**
+ * Represents a local query: queries with features implemented by the SDK on top
+ * of server implemented features.
+ */
 export class Query {
   static atPath(path: ResourcePath): Query {
-    return new Query(path);
+    return new Query(new Target(path));
```

Comment:
Per chat conversation, rather than have `Query` _wrap_ `Target` I think they should perhaps be 100% separate, with `Target` being a trivial class, perhaps something like:

```
export class Target {
  constructor(
    readonly path: ResourcePath,
    readonly collectionGroup: string | null = null,
    readonly explicitOrderBy: OrderBy[] = [],
    readonly filters: Filter[] = [],
    readonly limit: number | null = null,
    readonly startAt: Bound | null = null,
    readonly endAt: Bound | null = null
  ) { };
}
```

We might end up needing a few methods as well, but in particular, we should _not_ need to "build up" a Target by calling `.addFilter()`, `.withStartAt()`, etc.  Instead `Query.toTarget()` should just construct the `Target` on demand (or for perf it could create it once and then cache it if we expect toTarget() to be called repeatedly).

I don't feel super strongly about this, but I _think_ it'll be easier to reason about the types if we separate them completely (so `Query` isn't a wrapper) and make `Target` just a very simple type that's (basically) only used for serializing to/from the backend and persistence.